### DISCUSSION TITLE
Is there a binding to kill the entire buffer and not just a line?

### DISCUSSION THREAD

**Discussion by txtyash** [fish-shell/fish-shell#9165] [Q&A] [answered]
I have this binding: `bind -M $mode \cc end-of-buffer kill-whole-line repaint` which only removes single line in case I have multiple lines in my prompt.
```console
> long command \
        spans over multiple \
            lines|
```
If `|` represents the cursor then pressing `C-c` only deletes that line and not the contents of the entire buffer. Is there some other binding to do this?

**Comment by krobelus** [comment, marked as answer]
you can use this:

```fish
function my-cancel-commandline
    commandline ""
    emit fish_cancel
    # Repaint even if we haven't cancelled anything,
    # so the prompt refreshes and the terminal scrolls to it.
    commandline -f repaint
end
```

(maybe it's time to revive https://github.com/fish-shell/fish-shell/pull/4713)

**Reply by txtyash** [reply]
I'd rather use the function you gave me. Reviving old issues is like trying to bring people back from the dead.

**Comment by 1lessbug** [comment]
would control + U work for you? I'm trying to change binding to Command + Delete but can't find sequence for Command or Delete

**Reply by krobelus** [reply]
> I'm trying to change binding to Command + Delete but can't find sequence for Command or Delete

use `fish_key_reader`

**Reply by 1lessbug** [reply]
Nope doesn't read the command key

**Reply by txtyash** [reply]
> would control + U work for you? I'm trying to change binding to Command + Delete but can't find sequence for Command or Delete

control+U only deletes a single line. I wanted to delete multiple. The above function works fine for me.
